llvm/include/llvm/CodeGen/MachineInstr.h

```diff
@@ -248,15 +260,19 @@ class MachineInstr
     size_t numTrailingObjects(OverloadToken<uint32_t>) const {
       return HasCFIType;
     }
+    size_t numTrailingObjects(OverloadToken<MMRAMetadata>) const {
```

Comment:
I don't really grok how trailing objects are implemented, but this smells wrong to me. Shouldn't this function be folded into the overload for MDNode* just above? What makes this metadata different from the other metadata?

Reply:
This was just a leftover from the previous iteration of the design where I stored the MMRAMetadata object directly